lib/SPIRV/SPIRVRegularizeLLVM.cpp

```diff
@@ -98,8 +98,24 @@ class SPIRVRegularizeLLVM : public ModulePass {
   ///
   /// The actual implementation algorithm will be slightly different for
   /// simplification purposes.
-  void lowerFunnelShiftLeft(IntrinsicInst *FSHLIntrinsic);
-  void buildFunnelShiftLeftFunc(Function *FSHLFunc);
+  void buildFunnelShiftLeftFunc(Function *FSHRFunc);
+
+  /// No SPIR-V counterpart for @llvm.fshr.* intrinsic. It will be lowered
+  /// to a newly generated @spirv.llvm_fshr_* function.
+  ///
+  /// Conceptually, FSHR:
+  /// 1. concatenates the ints, the first one being the more significant;
+  /// 2. performs a right shift-rotate on the resulting doubled-sized int;
+  /// 3. returns the least significant bits of the shift-rotate result,
+  ///    the number of bits being equal to the size of the original integers.
+  /// If FSHR operates on a vector type instead, the same operations are
+  /// performed for each set of corresponding vector elements.
+  ///
+  /// The actual implementation algorithm will be slightly different for
+  /// simplification purposes.
+  void buildFunnelShiftRightFunc(Function *FSHRFunc);
+
+  void lowerFunnelShift(IntrinsicInst *FSHRIntrinsic);
```

Comment:
```suggestion
  void lowerFunnelShift(IntrinsicInst *FSHIntrinsic);
```
To match the naming at the definition's site.
I'd maybe even suggest renaming to `FunnelShiftIntrinsic` here and in the definition - while "FSHR" is an exact match for the intrinsic name, the meaning of "FSH" arguably cannot be grasped as quickly.